### PR TITLE
Fix auto semver tagging

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

- Fix auto semver tagging for auto generated git tags
- Triggered semver tagging for versions 1.0.7 and 1.0.9 that were missing due to the bug introduced because of the auto git tagging
- Driveby update to make semver job depend on the CI job to make sure semver tagging on main does not keep failing

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change

## How Has This Been Tested?

1.0.7 and 1.0.9 now in ECR:
image

image

## Reviewer Checklist

- [x] The PR represents a single feature (small drive-by fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [x] The code is sufficiently commented, particularly in hard-to-understand areas
- [x] Any required documentation updates have been made
- [x] Any TODOs added are captured in future tickets
- [x] No FIXMEs remain

### CODE REVIEW

File: .github/retag-and-push.sh [hunk 1]
```diff
@@ -52,6 +36,34 @@ process_tagged_version() {
...
 
+name=$(clean_string "${DOCKER_REGISTRY}/${project}")
+input_image="${project}:${image_tag}"
+
+if [[ -n "${NEW_TAG}" ]]; then
+    if is_valid_tag_name ${NEW_TAG} ; then
+        # push `semver` tagged image
+        semver="${NEW_TAG/v/}"
+        echo "Detected Tag: ${semver}"
+        process_tagged_version ${semver}
```
Comment (diversemix) [inline .github/retag-and-push.sh:47]: Do we want to `exit 0` here? ... If not - what is the reason to execute the code below? (sorry for the dumb question)

File: .github/workflows/main-push-autotag.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      # User controlled input needs to be santitised beforehand e.g., by adding an
+      # intermediate env var to prevent the workflow being exposed to a critical
+      # command injection attack
+      env:
+        PR_BODY: '${{ fromJson(steps.get_pr_data.outputs.result).body }}'
+        PR_NUMBER: '${{ fromJson(steps.get_pr_data.outputs.result).number }}'
+      run: |
+        echo "Attempting to auto-tag merge commit"
+        new_tag=$(.github/auto-tag.sh "${PR_BODY}" "${PR_NUMBER}")
+        echo "new_tag=${new_tag}"  >> "$GITHUB_OUTPUT"
```
Comment (diversemix) [inline .github/workflows/main-push-autotag.yml:42]: In the script its referenced in caps: `${NEW_TAG}`

Comment (katybaulch) [reply .github/workflows/main-push-autotag.yml:42]: See the `outputs` section of this job :)

**Review by diversemix — APPROVED** [review]
looks good - I think I follow this, just a question about the update to the push and retag script.

**Pull request merged**